app/components/UI/AssetOverview/AssetOverview.tsx

```diff
@@ -122,6 +127,32 @@ const AssetOverview: React.FC<AssetOverviewProps> = ({
     navigation.navigate('SendFlowView', {});
   };
 
+  const goToSwaps = () => {
+    navigation.navigate('Swaps', {
+      screen: 'SwapsAmountView',
+      params: {
+        sourceToken: swapsUtils.NATIVE_SWAPS_TOKEN_ADDRESS,
+        sourcePage: 'MainView',
+      },
+    });
+    trackEvent(MetaMetricsEvents.SWAP_BUTTON_CLICKED, {
+      text: 'Swap',
+      tokenSymbol: '',
+      location: 'TokenDetails',
+      chain_id: getDecimalChainId(chainId),
+    });
+  };
+  const onBuy = () => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    navigation.navigate(...createBuyNavigationDetails());
```

Comment:
```suggestion
    const [route, params] = createBuyNavigationDetails();
    navigation.navigate(route, params || {});
```

Will this allow us to avoid `@ts-ignore` here?